app/controllers/my-services/get-index-controller.js

```diff
@@ -29,32 +29,24 @@ module.exports = (req, res) => {
     return response(req, res, 'services/index', data)
   }
 
-  const reflectPromises = servicesRoles.map(serviceRole => {
-    return new Promise(function (resolve, reject) {
-      serviceService.getGatewayAccounts(serviceRole.service.gatewayAccountIds, req.correlationId)
-        .then(accounts => {
-          accounts.sort((a, b) => a.type === b.type ? 0 : a.type === 'live' ? -1 : 1)
-          const cardAccounts = accounts.filter(account => account.payment_method === undefined)
-          const directdebitAccounts = accounts.filter(account => account.payment_method === 'direct debit')
-          resolve({
-            name: serviceRole.service.name === 'System Generated' ? 'Temporary Service Name' : serviceRole.service.name,
-            external_id: serviceRole.service.externalId,
-            gateway_accounts: {
-              cardAccounts,
-              directdebitAccounts
-            },
-            permissions: getHeldPermissions(serviceRole.role.permissions.map(permission => permission.name))
-          })
-        })
-        .catch(() => reject(new Error('Error getting gateway account')))
+  const aggregatedGatewayAccountIds = servicesRoles
+    .map(servicesRole => servicesRole.service.gatewayAccountIds)
+    .reduce((acc, cv) => acc.concat(cv))
```

Comment:
is this known idiomatic js? I would prefer proper names `accumulator` `currentValue` - we don't have to optimise the size of our files these days